pkg/bid/bid.go

```diff
@@ -0,0 +1,149 @@
+package bid
+
+import (
+	"fmt"
+	"strings"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+)
+
+const (
+	BidPrefix            = "bid:"
+	ResourceBidPrefix    = BidPrefix + "r:"
+	EntitlementBidPrefix = BidPrefix + "e:"
+	GrantBidPrefix       = BidPrefix + "g:"
+)
+
+func ToBid(b interface{}) string {
+	switch bType := b.(type) {
+	case *v2.Resource:
+		return ResourceToBid(bType)
+	case *v2.Entitlement:
+		return EntitlementToBid(bType)
+	case *v2.Grant:
+		return GrantToBid(bType)
+	}
+	return "☠️" // TODO: panic? this should never happen
+}
+
+func ParseBid(bidStr string) (interface{}, error) {
```

Comment:
I wonder if we should make our own 'BID' type/proto that has a oneof for all the possible types?

If it was a oneof callers call 'ParseBID' and then get a concrete type back, then you'd call 'GetResource()' on it (or whatever) to get your parsed type out.

Or you can type switch on the oneof of you don't know what the id is, which is still a little cleaner than a bare interface because we know what types are allowed in the oneof.